zkevm-circuits/src/evm_circuit/util/math_gadget.rs

```diff
@@ -896,3 +901,255 @@ impl<F: Field> MulAddWordsGadget<F> {
         self.overflow.clone()
     }
 }
+
+#[derive(Clone, Debug)]
+pub(crate) struct ShrWordsGadget<F> {
+    a: util::Word<F>,
+    shift: util::Word<F>,
+    b: util::Word<F>,
+    // four 64-bit limbs of word `a`
+    a64s: [Cell<F>; 4],
+    // four 64-bit limbs of word `b`
+    b64s: [Cell<F>; 4],
+    // lower `n` bits
```

Comment:
You could use `shf_mod64` instead of `n` in the comments? I think it would make easier to understand.